staging/src/k8s.io/client-go/util/cert/cert.go

```diff
@@ -138,6 +138,34 @@ func MakeEllipticPrivateKeyPEM() ([]byte, error) {
 // Host may be an IP or a DNS name
 // You may also specify additional subject alt names (either ip or dns names) for the certificate
 func GenerateSelfSignedCertKey(host string, alternateIPs []net.IP, alternateDNS []string) ([]byte, []byte, error) {
+	caKey, err := rsa.GenerateKey(cryptorand.Reader, 2048)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	caTemplate := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			CommonName: fmt.Sprintf("%s-ca@%d", host, time.Now().Unix()),
+		},
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().Add(time.Hour * 24 * 365),
+
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+		IsCA: true,
```

Comment:
Do we want to switch IsCA on line 185 to false (the one github won't let me comment on...)  Or are we worried that might be a breaking change?

(And have we checked that the problematic curls tolerate this cert chain?)

Reply:
> You could avoid generating an additional cert if pathLenConstraint is 0, as it should remove the ability of any signed cert to sign certs that can be validated by the cert chain.

I want to avoid the serving cert being able to be used as a client cert as well, which it could be with all ext key usages removed.

> Do we want to switch IsCA on line 185 to false (the one github won't let me comment on...)

yeah, forgot to drop that. done.

> Or are we worried that might be a breaking change?

anyone referencing the cert file as a trust root bundle will still pick up the CA with this change, so we're good.

> have we checked that the problematic curls tolerate this cert chain?

yes. before:
```sh
$ curl https://localhost:6443/api --cacert ./apiserver.crt
curl: (60) Certificate type not approved for application.
```

after:
```sh
$ curl https://10.192.208.6:6443/api --cacert ./apiserver.crt 
{
  "kind": "APIVersions",
  "versions": [
    "v1"
  ],
  "serverAddressByClientCIDRs": [
    {
      "clientCIDR": "0.0.0.0/0",
      "serverAddress": "localhost:6443"
    }
  ]
}
```

Reply:
@liggitt , i've just filed https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/557, and am wondering whether or not we should do the same here and found this. Is there a reason not to set pathLenConstraint?

Reply:
> Is there a reason not to set pathLenConstraint?

is there a reason to, if we only generate a single serving cert and then throw away the CA signing key?